### PR TITLE
binutils: add variant 'nls' for native language support

### PR DESCRIPTION
**Pull request by mwkrentel**
Add variant 'nls' for native language support with default True.
The default inside binutils was always on, but this gives a way of
turning it off, if desired.

Adjust the dependencies.  Flex is never used for a one-time build.
Bison and m4 should not be needed, except that prior to rev 2.30, gold
did not include its generated files, so bison is needed when +gold.

Drop configure option --enable-interwork.  This option does not exist.

### CODE REVIEW

File: var/spack/repos/builtin/packages/binutils/package.py [hunk 1]
```diff
@@ -45,29 +45,30 @@ class Binutils(AutotoolsPackage):
...
     patch('update_symbol-2.26.patch', when='@2.26')
 
     depends_on('zlib')
+    depends_on('gettext', when='+nls')
 
-    depends_on('m4', type='build')
-    depends_on('flex', type='build')
-    depends_on('bison', type='build')
-    depends_on('gettext')
+    # Prior to 2.30, gold did not distribute the generated files and
```
Comment (healther) [inline var/spack/repos/builtin/packages/binutils/package.py]: if this is only needed for :2.30, then you can use `when='@:2.30 +gold'`

**Comment by mwkrentel** [comment]
That's right.  But it's only two data points, 2.30 and 2.31 that now
include the generated files.  I was concerned about drawing a firm
conclusion based on too few data points and I didn't want to break
anyone else's use case going forward.

But I could try a few more tar files.  You think that would be
preferable to add '@:2.30'?

So, '@:2.30' means inclusive comparison, right?
That is, it means <= 2.30, not strictly < 2.30 ?

In that case, I should use '@:2.29' or '@:2.29.1'.

**Comment by mwkrentel** [comment]
Ok, I changed depends_on() to when='@:2.29.99 +gold' and I tested both
sides of the line.

If this changes in a future rev, then it's somebody else's fault,
not mine, right?   :-)

I'm done.  Anything else?

**Comment by scheibelp** [comment]
> Is codecov in a wonky state or something?

It should report the line coverage. However, for package PRs you don't have to worry about coverage.

@healther thanks for reviewing!

**Comment by healther** [comment]
> Is codecov in a wonky state or something?

@mwkrentel Just as a technical explanation, if you are curious: Codecov checks the state of coverage of the HEAD of this PR, so all commits in this PR based on the mainline commit, that you branched of from, against the coverage of the current HEAD of develop. And it only checks core spack, it doesn't check any `package.py`.
If now there is a merge of a PR that increases coverage on `develop`, codecov will notice that this HEAD has less coverage than `develop` (although this will be fixed once it is merged). The "correct" technical solution would be to do automatic rebases, but it's much easier to just not care for package only changes ;)

**Comment by mwkrentel** [comment]
@healther  Yes, I am interested, and yes, that was helpful.

So, I could submit a PR, get a green check, wait while someone else
makes commits to develop, make a trivial change to my PR and get a
totally different answer from codecov?  That does at least explain
some of the behavior I've seen.

If this happens again and I get a red X that I feel is undeserved,
then I'll just do the rebase myself, see if I get a different result.

I don't know if this is in the docs, but I suggest it should be, along
with the suggestion of rebasing.

Thanks!

**Comment by healther** [comment]
> So, I could submit a PR, get a green check, wait while someone else
makes commits to develop, make a trivial change to my PR and get a
totally different answer from codecov? 

Yes

> If this happens again and I get a red X that I feel is undeserved,
then I'll just do the rebase myself, see if I get a different result.

Don't bother, at least for package-only PRs. 

> I don't know if this is in the docs, but I suggest it should be, along
with the suggestion of rebasing.

There should be the information, that we only require the flake8 tests to pass. But I haven't looked at the docs in detail in quite some time.

**Pull request merged**